internal/pod_models.go

```diff
@@ -554,6 +554,14 @@ func configureSidecarContainer(container *corev1.Container, initMode bool, proce
 
 		sidecarEnv = append(sidecarEnv, getEnvForMonitorConfigSubstitution(cluster, processGroupID)...)
 
+		if cluster.UseDNSInClusterFile() {
+			sidecarArgs = append(sidecarArgs, "--substitute-variable", "FDB_DNS_NAME")
+		}
+
+		if cluster.UseDNSInClusterFile() {
+			sidecarEnv = append(sidecarEnv, corev1.EnvVar{Name: "FDB_DNS_NAME", Value: GetPodDNSName(cluster, podName)})
+		}
+
```

Comment:
```suggestion
		if cluster.UseDNSInClusterFile() {
			sidecarArgs = append(sidecarArgs, "--substitute-variable", "FDB_DNS_NAME")
			sidecarEnv = append(sidecarEnv, corev1.EnvVar{Name: "FDB_DNS_NAME", Value: GetPodDNSName(cluster, podName)})
		}

```

In a follow up PR wee probably want to move all of those env strings into constants.